### PR TITLE
Barista/improve toc

### PR DESCRIPTION
**Pull request by lara-aigmueller**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/barista/src/pages/page-outlet.ts [hunk 1]
```diff
@@ -97,6 +97,11 @@ export class BaPageOutlet {
     );
 
     this._currentPageComponentRef = componentRef;
+
+    // Reset scroll position on new page load.
+    window.scrollTo({
```
Comment (thomaspink) [inline apps/barista/src/pages/page-outlet.ts]: Would like to see a check for window here.

File: apps/barista/src/shared/toc.service.ts [hunk 2]
```diff
@@ -70,9 +70,19 @@ export class BaTocService {
 
     // start the scroll spy
     this._scrollSpyInfo = this._scrollSpyService.spyOn(headlines);
-    this._scrollSpyInfo.active.subscribe(item => {
-      if (item) {
-        this.activeItem.next(item.element);
+    this._scrollSpyInfo.active.subscribe(scrollItem => {
```
Comment (thomaspink) [inline apps/barista/src/shared/toc.service.ts]: Pls unsubscribe

File: apps/barista/src/layout/toc/toc.ts [hunk 3]
```diff
@@ -26,64 +25,33 @@ import { BaTocService } from '../../shared/toc.service';
...
+  ngOnInit(): void {
+    this._tocService.tocList.subscribe(headings => {
+      this._headings = headings;
+    });
 
-  constructor(private tocService: BaTocService) {
-    const docElement = document.getElementById('all-content') || undefined;
-    this.tocService.genToc(docElement);
-    this._headings = this.tocService.tocItems;
+    this._tocService.activeItems.subscribe(activeItems => {
```
Comment (thomaspink) [inline apps/barista/src/layout/toc/toc.ts]: Pls unsubscribe

File: apps/barista/src/layout/toc/toc.ts [hunk 4]
```diff
@@ -26,64 +25,33 @@ import { BaTocService } from '../../shared/toc.service';
...
+  _headings: BaTocItem[] = [];
+  /** @internal the toc entries that are currently active */
+  _activeItems: BaTocItem[] = [];
+
+  constructor(private _tocService: BaTocService) {}
 
-  /** the toc entry that is currently active */
-  private _activeElement;
+  ngOnInit(): void {
+    this._tocService.tocList.subscribe(headings => {
```
Comment (thomaspink) [inline apps/barista/src/layout/toc/toc.ts]: Pls unsubscribe

**Comment by thomaspink** [comment]
LGTM

**Comment by lara-aigmueller** [comment]
> LGTM

had to rebase, pls review again.

**Pull request merged**